compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaSingletonTypeSymbol.java

```diff
@@ -48,6 +48,8 @@ public BallerinaSingletonTypeSymbol(CompilerContext context, ModuleID moduleID,
             this.typeName = "()";
         } else if (shape.getOriginalValue() != null) {
             this.typeName = shape.getOriginalValue();
+        } else if (shape.getValue() == null) {
+            this.typeName = "";
```

Comment:
Instead of rechecking for null here, can modify the conditions in lines 46 and 47 to incorporate this can't we?